### DISCUSSION TITLE
Error create client creating connection: making auth key: handshake: Can't find fingerprint

### DISCUSSION THREAD

**Discussion by muhsinshodiq** [xelaj/mtproto#106] [Q&A]
Hi All,

Need help, I'm trying to create telegram client as below:

```go
_, err := telegram.NewClient(telegram.ClientConfig{
	AppID:           appID,
	AppHash:         appHash,
	ServerHost:      server,
	PublicKeysFile:  key,
	SessionFile:     "session_file.json",
	InitWarnChannel: true,
})

if err != nil {
	fmt.Println("Error create client", err)
	return
}
```

But it always get a result:
`Error create client creating connection: making auth key: handshake: Can't find fingerprint`

I got public keys through [https://my.telegram.org](https://my.telegram.org) what I'm missing about?

**Comment by quenbyako** [comment]
Hey @muhsinshodiq, thanks for asking!

This looks REALLY strange issue, cause, if you try to find [this error text](https://github.com/xelaj/mtproto/blob/04ee89d1b6bce00fb50e4e87bf16fdc781cf4d8f/handshake.go#L45) telegram server _literally must_ return at least one fingerprint...

I'll try to figure it out, what's happened. But you should notice, that i have no any contact with telegram devs, so i can't guarantee that i find any fix for this issue...

**Reply by muhsinshodiq** [reply]
Hi @quenbyako thx for response, I'm trying to modified the code and check is fingerprint is returned or not, and When do debugging on `handshake.go` is returned one fingerprints, but the value is not match as described on 2nd picture.

![image](https://user-images.githubusercontent.com/16998592/128448197-eaf16d23-ad77-4593-b522-25ef6c22fe8e.png)

![image](https://user-images.githubusercontent.com/16998592/128448141-b3f006d1-da4e-47c4-b7a2-7b0ff85a234c.png)

**Reply by quenbyako** [reply]
are you sure that your connection wasn't compromised? Cause, well, maybe.... Few countries really don't like telegram, so they trying to use some mitm hacks, so, are you sure, that this is not your situation?

**Reply by muhsinshodiq** [reply]
@quenbyako I'm trying using two internet provider in my country and also trying to use VPN from my office and the result still the same like above.

I store public key that I got from telegram into `.pem` extension and point the config to this file, is it correct?

**Comment by vpoltora** [comment]
Good afternoon! A similar problem
telegram gave 2 keys, I save them to the .pem file, specify the rest of the parameters and the same error is obtained.
If I use your accesses obtained through utils.PrepareAppStorageForExamples (), everything is fine, authentication succeeds. I change to my own - error :(

**Reply by vpoltora** [reply]
@muhsinshodiq did you find a solution?

**Reply by muhsinshodiq** [reply]
@itxor not yet, still working and trying to find out what the issue.

**Reply by waybackarchiver** [reply]
I found that replacing the `app_id` and `app_hash` worked in [`example/auth`](https://github.com/xelaj/mtproto/blob/5eb6237dcf631e69d7694418990b32b333f99586/examples/auth/main.go), but porting the code to my project caused the same error.

**Comment by Manuele99** [comment]
Hi, I've been hit by the same issue.
When I registered my application I received 2 public keys.
I've tried to use them singularly, one with production IP and one with test ip, none of that worked. I've also tried to put them all in the same .pem file, like 
```
-----BEGIN RSA PUBLIC KEY-----
XXX
-----END RSA PUBLIC KEY-----

-----BEGIN RSA PUBLIC KEY-----
XXX
-----END RSA PUBLIC KEY-----
```
but nothing.

The only way I'm able to get this working is by using the manually downloaded [keys](https://github.com/xelaj/mtproto/blob/main/examples/example_utils/main.go#L20) from the example with `wget -O pub_keys.pem https://git.io/JtImk`.

That's the furthest point I've come to debug.
Could it be a problem on the telegram API side?

**Reply by koaim** [reply]
+1

**Comment by coderisimo** [comment]
I have the same trouble :( Any updates ?

**Reply by quenbyako** [reply]
I have some but i need a help. @coderisimo please, if you have a free time, may i ask you for contributing in [tl package](https://github.com/xelaj/tl)? Last task is related to optimize encoder and decoder registry, and after that we will completely rework mtproto

**Comment by TinGimranov** [comment]
Has anybody solved this problem? I have the same.

**Comment by ruggero118** [comment]
Same problem. Any solution/work around to get this implementation working?
